### PR TITLE
improved valdation while parsing the secret

### LINKED ISSUE #59: Warn if key is used several times in a secret.
When a key is used in the same secret several times only the value of the latter key will be saved. For example:
```
password: test
password: test2
```
The key "password" will have the value "test2".

It would be nice if the user gets a warning and the chance to re-edit the secret.

### PR DESCRIPTION
**Pull request by winpat**
Improved parsing of the secret, so that a key (identifier) can't be used multiple times. If a key is
found multiple times, the user will be asked to re-edit the secret.

Fixes #59

### CODE REVIEW

File: src/edit.go [hunk 1]
```diff
@@ -46,16 +53,34 @@ func (c *EditCommand) Run(args []string) int {
...
+
+		err = EditFile(file.Name())
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Unable to edit secret file %q", err))
+			return 1
+		}
+
+		data, err = ParseSecret(file.Name())
+		switch err {
+		case ErrDuplicateKey:
```
Comment (sliverc) [inline src/edit.go]: To avoid duplicate code you can list cases comma separated.
case ErrDuplicateKey, ErrMultipleDelimiters, ErrMissingDelimiter:

File: src/edit.go [hunk 2]
```diff
@@ -116,49 +141,74 @@ func EditFile(path string) error {
...
 
 		if line != "" {
+
 			if strings.HasPrefix(line, "#") {
-				// If line is a comment, store it's value for later composition into it's
-				// own k/v pair
+
 				if comment != "" {
-					// If a comment is alreay set, then assume that the comment spans
-					// across multiple lines
```
Comment (sliverc) [inline src/edit.go]: what's the reason that you removed this comment?

File: src/edit.go [hunk 3]
```diff
@@ -116,49 +141,74 @@ func EditFile(path string) error {
...
 					comment = strings.TrimPrefix(line, "#")
 				}
 
 			} else {
-				// If a line is a k/v pair then split it up
+
 				kv_pair := strings.Split(line, ": ")
+				if len(kv_pair) < 2 {
+					fmt.Printf("Unable to parse key/value pair %q. Make sure that there is at least one \": \" delimiter in it ", line)
+					_, _, _ = bufio.NewReader(os.Stdin).ReadLine()
```
Comment (sliverc) [inline src/edit.go]: Why is this needed? Maybe a comment would be good. Also as it happens on every return might better happen in a defer statement.

File: src/edit.go [hunk 4]
```diff
@@ -116,49 +141,74 @@ func EditFile(path string) error {
...
 				} else {
 					comment = strings.TrimPrefix(line, "#")
 				}
 
 			} else {
-				// If a line is a k/v pair then split it up
+
 				kv_pair := strings.Split(line, ": ")
+				if len(kv_pair) < 2 {
+					fmt.Printf("Unable to parse key/value pair %q. Make sure that there is at least one \": \" delimiter in it ", line)
```
Comment (sliverc) [inline src/edit.go]: Have you considered using stderr here? This question includes the other error prints below.

File: src/edit.go (hunk 1 above)
Comment (winpat) [reply src/edit.go]: Thanks! I tried to lookup on how to "or"-connect the errors into one statement. However I could not figure it out.  That's exactly what I was looking for.

File: src/edit.go (hunk 2 above)
Comment (winpat) [reply src/edit.go]: Probably got cut off while refactoring. I will re-add it.

File: src/edit.go (hunk 4 above)
Comment (winpat) [reply src/edit.go]: Great input. I haven't but it would be much cleaner.

**Comment by sliverc** [comment]
LGTM now
